apps/jobs/views.py

```diff
@@ -958,6 +959,71 @@ def update_submission(request, challenge_pk):
         return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
 
 
+@api_view(['POST', ])
+@throttle_classes([UserRateThrottle, ])
+@permission_classes((permissions.IsAuthenticated, HasVerifiedEmail))
+@authentication_classes((ExpiringTokenAuthentication,))
+def re_run_submission(request, submission_id):
+    """
+    API endpoint to re-run a submission.
+    Only challenge host has access to this endpoint.
+    """
+    try:
+        submission = Submission.objects.get(id=submission_id)
+        logger.info('Submission found with submission number {}'
+                    .format(submission_id))
+    except Submission.DoesNotExist:
+        response_data = {'error': 'Submission with submission number {} does not exist'.format(submission_id)}
+        return Response(response_data, status=status.HTTP_404_NOT_FOUND)
+
+    # check if the challenge phase exists or not
+    try:
+        challenge_phase = submission.challenge_phase
+        challenge_phase_id = challenge_phase.id
+    except Exception as e:
+        response_data = {'error': 'Challenge Phase does not exist'}
+        return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
+
+    # check if the challenge exists or not
+    try:
+        challenge = challenge_phase.challenge
+        challenge_id = challenge.id
+    except Exception as e:
+        response_data = {'error': 'Challenge does not exist'}
+        return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
+
+    if not is_user_a_host_of_challenge(request.user, challenge_id):
+        response_data = {
+            "error": "You are not allowed to re-run the challenge submission"
+        }
+        return Response(response_data, status=status.HTTP_403_FORBIDDEN)
+
+    if not challenge.is_active:
+        response_data = {'error': 'Challenge is not active'}
+        return Response(response_data, status=status.HTTP_406_NOT_ACCEPTABLE)
+
+    # check if challenge phase is active
+    if not challenge_phase.is_active:
+        response_data = {
+            'error': 'Sorry, cannot accept submissions since challenge phase is not active'}
+        return Response(response_data, status=status.HTTP_406_NOT_ACCEPTABLE)
+    try:
+        publish_submission_message(challenge_id, challenge_phase_id, submission.id)
+        time.sleep(1)
+        submission = Submission.objects.get(id=submission_id)
+        response_data = {
+            'submission_status': submission.status,
+            'submission_execution_time': submission.execution_time,
+            'submission_number': submission.submission_number,
+            'success': 'Submission result has been successfully updated'
```

Comment:
'Submission result has been successfully updated' -> `Submission is successfully submitted for re-running`